core/station/impl/src/services/cycle_manager.rs

```diff
@@ -0,0 +1,102 @@
+use crate::factories::blockchains::InternetComputer;
+use crate::models::CycleObtainStrategy;
+use canfund::api::cmc::IcCyclesMintingCanister;
+use canfund::api::ledger::IcLedgerCanister;
+use canfund::manager::options::{FundManagerOptions, FundStrategy, ObtainCyclesOptions};
+use canfund::manager::RegisterOpts;
+use canfund::operations::obtain::MintCycles;
+use canfund::FundManager;
+use ic_cdk::api::management_canister::main::CanisterId;
+use ic_ledger_types::{Subaccount, MAINNET_CYCLES_MINTING_CANISTER_ID, MAINNET_LEDGER_CANISTER_ID};
+use lazy_static::lazy_static;
+use std::cell::RefCell;
+use std::sync::Arc;
+
+thread_local! {
+    static FUND_MANAGER: RefCell<FundManager> = RefCell::new(FundManager::new());
+}
+
+lazy_static! {
+    pub static ref CYCLE_MANAGER: Arc<CycleManager> = Arc::new(CycleManager::new());
+}
+
+#[derive(Debug, Default)]
+pub struct CycleManager {}
+
+impl CycleManager {
+    fn new() -> Self {
+        FUND_MANAGER.with(|manager| {
+            let mut manager = manager.borrow_mut();
+            // Strategy can be default as we always override per canister
+            // Obtain cycles config is inherited from the system service
+            let options = FundManagerOptions::new().with_interval_secs(60 * 60 * 6);
```

Comment:
nit: make a note here that this interval of 6 hours is used for all monitored canisters